receiver/kubeletstatsreceiver/factory.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubeletstatsreceiver
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+	"go.uber.org/zap"
+)
+
+const typeStr = "kubeletstats"
+
+var _ component.ReceiverFactoryBase = (*Factory)(nil)
+
+type Factory struct {
+}
+
+func (f Factory) Type() configmodels.Type {
+	return typeStr
+}
+
+func (f Factory) CreateDefaultConfig() configmodels.Receiver {
+	return &Config{
+		ReceiverSettings: configmodels.ReceiverSettings{
+			TypeVal: typeStr,
+		},
+	}
+}
+
+func (f Factory) CustomUnmarshaler() component.CustomUnmarshaler {
+	return nil
+}
+
+func (f Factory) CreateTraceReceiver(
+	ctx context.Context,
+	logger *zap.Logger,
+	cfg configmodels.Receiver,
+	nextConsumer consumer.TraceConsumerOld,
+) (component.TraceReceiver, error) {
+	return nil, nil
```

Comment:
This should return `configerror.ErrDataTypeIsNotSupported`.

Reply:
Fixed.